### DISCUSSION TITLE
Question on determining a font's cap height

### DISCUSSION THREAD

**Discussion by nickjg231** [uswds/uswds#5546] [Q&A] [answered]
Hello USWDS team,

I'm a front-end developer working on the VA's [vets-website](https://github.com/department-of-veterans-affairs/vets-website) repository and a few other repositories. We use your font normalization in our project, and we're trying to add the [Bitter font](https://fonts.google.com/specimen/Bitter) to our own project. 

I've been diving into the typography setup within the USWDS and came across the cap-height values introduced in [commit 8cb2e64](https://github.com/uswds/uswds/commit/8cb2e644764b6200f321359635207712c8b8becc) by @mejiaj. The specific pixel values for each font's cap height have caught my attention.

Could anyone please shed some light on the methodology or tools used to determine these values? Were they derived from a font editor, a typographic tool, or manual measurements? Additionally, I noticed that the values are all within the range between 340-380px — is there a specific rationale behind this or was it a coincidence based on the fonts in use?

Understanding the process would be invaluable for those of us looking to extend or adapt the framework with new fonts, ensuring consistency in our implementations.

Thank you for your time and all the hard work on USWDS!

**Comment by caw310** [comment]
@mejiaj could you provide some help on this?

**Comment by mejiaj** [comment]
Hello! I'm not as familiar as @thisisdano on font normalization, but I'll do my best to help.  

The normalization settings mentioned in the commit were transferred over from USWDS 2.x.

>Could anyone please shed some light on the methodology or tools used to determine these values? Were they derived from a font editor, a typographic tool, or manual measurements? Additionally, I noticed that the values are all within the range between 340-380px — is there a specific rationale behind this or was it a coincidence based on the fonts in use?

The cap-height looks to be a manual value. The default is `364px`, which is shown in the code example in guidance [Adding a font from a hosting service](https://designsystem.digital.gov/design-tokens/typesetting/font-family/#adding-fonts-to-uswds-2). The value comes from comparing against the height of the `N` character (`San Francisco` & `Roboto`) set at 500px [^1][^2].

[^1]: https://github.com/uswds/uswds/commit/4073851acd8e2c14448c7dd0a4059655fa68863d#
[^2]: https://designsystem.digital.gov/design-tokens/typesetting/overview/#fonts-with-normalization-applied-2

---

### For discussion

Should we use [font-size-adjust with `cap-height` [MDN]](https://developer.mozilla.org/en-US/docs/Web/CSS/font-size-adjust#cap-height) specified?

**Reply by babsdenney** [reply]
Thank you for your response! We were able to determine the cap-height of the Bitter font based on your guidance and also submitted an issue for some of the cap-heights being used for USWDS. 

[Cap-heights issue submitted](https://github.com/uswds/uswds/issues/5568)

**Comment by thisisdano** [comment, marked as answer]
Thanks for the help here @mejiaj — and you're right, the `cap-height` is the actual height of the vertical bar of the `N` character set to 500px. This definition used to be in our settings files, but we no longer distribute settings files in that way, so that important comment went away.

And really, the important thing is that this measure is approximate. All fonts have their own idiosyncrasies, and font normalization is only intended to do a better job easing the transition from font to font. It isn't perfect or exact, and it can't be, really.

**Reply by thisisdano** [reply]
Also, for what it's worth, we did our original measurements in Sketch. There may be some variation from application to application, OS to OS.

**Reply by caw310** [reply]
Thanks for the additional informaiton.

**Comment by ryparker** [comment]
The author confirmed that the answer explained how to measure cap height, so I'm closing that question as resolved. The [accepted answer](https://github.com/uswds/uswds/discussions/5546#discussioncomment-7318222) remains available. The separate concern about cap-height accuracy was tracked in #5568; this closure doesn't mean that concern was fixed.
